src/main/java/io/appform/jsonrules/expressions/JsonPathBasedExpression.java

```diff
@@ -68,5 +63,22 @@ public final boolean evaluate(ExpressionEvaluationContext context) {
         return evaluate(context, path, evaluatedNode);
     }
 
+    private JsonNode applyPreoperation(ExpressionEvaluationContext globalContext) {
+        JsonNode payload = globalContext.getNode().at(path);
+
+        if (null == preoperation) {
+            return payload;
+        }
+
+        ExpressionEvaluationContext nodeEvaluationContext = ExpressionEvaluationContext.builder()
```

Comment:
I'm not sure creating a context from the context here is a good idea. I foresee that at a later point in time, please will add stuff to the context and forget to add it to the copy here. Why is a copy required here? Also, even if this is the proper way to go about it, create a clone/deepCopy method in context and use that to create this copy. So that the logic of creating a clone is localized to context itself.